ADRs/0011-Python-Path-Resolution.md

```diff
@@ -0,0 +1,631 @@
+        > MarkupSafe | 1.1.1
+        > matplotlib | 3.2.1
+        > mccabe | 0.6.1
+        > mock | 4.0.1
+        > networkx | 1.9
+        > numpy @ file:///C:/Users/shams/Downloads/numpy-1.19.1%2Bmkl-cp37-cp37m-win_amd64.whl
+        > oauthlib | 3.1.0
+        > opencv-python | 4.2.0.32
+        > opt-einsum | 3.2.0
+        > packaging | 20.4
+        > pandas | 0.24.2
+        > pathspec | 0.7.0
+        > pep8 | 1.7.1
+        > Pillow | 7.0.0
+        > pkginfo | 1.5.0.1
+        > protobuf | 3.11.3
+        > py4j | 0.10.7
+        > pyarrow | 0.13.0
+        > pyasn1 | 0.4.8
+        > pyasn1-modules | 0.2.8
+        > pycodestyle | 2.5.0
+        > pydatavec | 0.1.2
+        > pydl4j | 0.1.5
+        > Pygments | 2.6.1
+        > pyjnius | 1.3.0
+        > pylint | 2.4.3
+        > pymongo | 3.11.0
+        > pyparsing | 2.4.7
+        > pyspark | 2.4.5
+        > pytest-cache | 1.0
+        > pytest-cov | 2.8.1
+        > pytest-pep8 | 1.0.6
+        > python-dateutil | 2.8.1
+        > pytz | 2020.1
+        > pywin32-ctypes | 0.2.0
+        > PyYAML | 5.3.1
+        > readme-renderer | 25.0
+        > regex | 2020.2.20
+        > requests | 2.24.0
+        > requests-oauthlib | 1.3.0
+        > requests-toolbelt | 0.9.1
+        > rsa | 4.0
+        > scikit-learn | 0.22.2.post1
+        > scipy | 1.4.1
+        > setupext-janitor | 1.1.2
+        > six | 1.15.0
+        > tensorboard | 2.1.1
+        > tensorflow | 2.1.0
+        > tensorflow-estimator | 2.1.0
+        > termcolor | 1.1.0
+        > toml | 0.10.0
+        > torch | 1.4.0+cpu
+        > torchvision | 0.5.0+cpu
+        > tqdm | 4.45.0
+        > twine | 3.1.1
+        > typed-ast | 1.4.0
+        > urllib3 | 1.25.9
+        > webencodings | 0.5.1
+        > Werkzeug | 1.0.0
+        > wrapt | 1.11.2
+        > zipp | 3.1.0
+.
+.
+.
+
+-   id: 3   
+    path: C:\Users\shams\AppData\Local\Microsoft\WindowsApps\python.exe                        
+    version: 3.6.1
+    .
+    .
+    .
+
+-----------------------------------------------------------------------------------------------------
+
+.
+.
+.
+
+```
+
+### Example Workflow
+Here is an example workflow that involves the actual API code snippets:
+
+#### 1. For Regular Python Install
+Assuming `konduit pythonpaths`, gives us the following information on a particular system for regular python installs:
+
+```text
+konduit-pc ~ % konduit pythonpaths
+
+-------------------------------------------PYTHON INSTALLS-------------------------------------------
+.
+.
+.
+
+-   id: 4
+    path: C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe
+    version: 3.7.8
+    packages: 
+    .
+    .
+    .
+
+.
+.
+.
+```
+
+Now we want to specify the python install with `id = 4` into the PythonStep configuration in Java. The code would look similar to: 
+
+```java
+PythonConfig pythonConfig = PythonConfig.builder()
+        .pythonType(PythonConfig.PythonType.Python)
+        .pythonPath("4")
```

Comment:
We probably should give the user the option to name the pipeline. IDs will be hard to remember.